Test failure caused by conflicting dependency versions?

Hiya! I'm still working on getting tests cleanly passing (there is at least one more PR I think I'll open for the `CONTRIBUTING.md` file), but I'm struggling a bit more with this one:

```
_________________________________________________________________________________ test_plot_network_layouts _________________________________________________________________________________

mock_show = <MagicMock name='show' id='140538206253232'>, mock_enable = <MagicMock name='output_notebook' id='140538206250640'>
evo_test_networks = (<networkx.classes.digraph.DiGraph object at 0x7fd19bde5040>, {'species1': <networkx.classes.digraph.DiGraph object at...es.digraph.DiGraph object at 0x7fd199dc7b30>, 'species3': <networkx.classes.digraph.DiGraph object at 0x7fd199dc6030>})

    @patch('bokeh.io.output_notebook')
    @patch('bokeh.plotting.show')
    def test_plot_network_layouts(mock_show, mock_enable, evo_test_networks):
        """Test different layout options with fallback handling"""
        # Test kamada_kawai and spring layouts which don't require external dependencies
        safe_layouts = ['kamada_kawai', 'spring']
        main_net, _ = evo_test_networks
        for layout in safe_layouts:
            plot = plot_network(main_net, plot_format=layout)
            assert plot is not None
            plt.close()
        # Test pydot2 layout with proper error handling
        try:
            with suppress_pydot_warnings():
>               plot = plot_network(main_net, plot_format='pydot2')
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

test_core_functions.py:5340:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../glycowork/network/biosynthesis.py:533: in plot_network
    pos = {'pydot2': lambda: nx.nx_pydot.pydot_layout(network, prog = "dot"),
../glycowork/network/biosynthesis.py:533: in <lambda>
    pos = {'pydot2': lambda: nx.nx_pydot.pydot_layout(network, prog = "dot"),
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

G = <networkx.classes.digraph.DiGraph object at 0x7fd19bde5040>, prog = 'dot', root = None

    def pydot_layout(G, prog="neato", root=None):
        """Create node positions using :mod:`pydot` and Graphviz.

        Parameters
        ----------
        G : Graph
            NetworkX graph to be laid out.
        prog : string  (default: 'neato')
            Name of the GraphViz command to use for layout.
            Options depend on GraphViz version but may include:
            'dot', 'twopi', 'fdp', 'sfdp', 'circo'
        root : Node from G or None (default: None)
            The node of G from which to start some layout algorithms.

        Returns
        -------
        dict
            Dictionary of positions keyed by node.

        Examples
        --------
        >>> G = nx.complete_graph(4)
        >>> pos = nx.nx_pydot.pydot_layout(G)
        >>> pos = nx.nx_pydot.pydot_layout(G, prog="dot")

        Notes
        -----
        If you use complex node objects, they may have the same string
        representation and GraphViz could treat them as the same node.
        The layout may assign both nodes a single location. See Issue #1568
        If this occurs in your case, consider relabeling the nodes just
        for the layout computation using something similar to::

            H = nx.convert_node_labels_to_integers(G, label_attribute="node_label")
            H_layout = nx.nx_pydot.pydot_layout(H, prog="dot")
            G_layout = {H.nodes[n]["node_label"]: p for n, p in H_layout.items()}

        """
        import pydot

        P = to_pydot(G)
        if root is not None:
            P.set("root", str(root))

        # List of low-level bytes comprising a string in the dot language converted
        # from the passed graph with the passed external GraphViz command.
        D_bytes = P.create_dot(prog=prog)

        # Unique string decoded from these bytes with the preferred locale encoding
        D = str(D_bytes, encoding=getpreferredencoding())

        if D == "":  # no data returned
            print(f"Graphviz layout with {prog} failed")
            print()
            print("To debug what happened try:")
            print("P = nx.nx_pydot.to_pydot(G)")
            print('P.write_dot("file.dot")')
            print(f"And then run {prog} on file.dot")
            return

        # List of one or more "pydot.Dot" instances deserialized from this string.
        Q_list = pydot.graph_from_dot_data(D)
        assert len(Q_list) == 1

        # The first and only such instance, as guaranteed by the above assertion.
        Q = Q_list[0]

        node_pos = {}
        for n in G.nodes():
            str_n = str(n)
>           node = Q.get_node(pydot.quote_id_if_necessary(str_n))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
E           AttributeError: module 'pydot' has no attribute 'quote_id_if_necessary'. Did you mean: 'quote_if_necessary'?

../.venv/lib/python3.12/site-packages/networkx/drawing/nx_pydot.py:353: AttributeError
================================================================================== short test summary info ==================================================================================
FAILED test_core_functions.py::test_plot_network_layouts - AttributeError: module 'pydot' has no attribute 'quote_id_if_necessary'. Did you mean: 'quote_if_necessary'?
```

If I manually install a newer version of `pydot` (`v4.0.1`, for example), then the tests pass, but that's not something I can add to the `pyproject.toml` file, because `glyles` requires `pydot<2.0.0`... We could maybe patch `glyles` to use a newer version, or trying to find some hack for forcing that newer `pydot` version (though that looks like we'll have to change how `glycowork` is installed and do something a bit messy like this: https://stackoverflow.com/questions/23969793/setup-py-pip-override-one-of-the-dependencys-sub-dependency-from-requirement).

I can reproduce the issue by running these commands:
```
cd /tmp
git clone https://github.com/BojarLab/glycowork.git
cd glycowork
uv venv
uv pip install -e ".[all,dev]"
cd tests
uv run pytest
```

Personally I think it might be best to update `glyles`, but I'm curious how others have been avoiding this failure until now?

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -42,7 +42,7 @@ all = [
     "torch_geometric",
     "torch",
     "xgboost",
-    "glyles",
+    "glyles~=1.2.3a0",
     "pubchempy",
     "requests",
     "py3Dmol"
@@ -52,7 +52,7 @@ dev = [
     "torch_geometric",
     "torch",
     "xgboost",
-    "glyles",
+    "glyles~=1.2.3a0",
     "pubchempy",
     "requests",
     "py3Dmol",
@@ -64,7 +64,7 @@ ml = [
     "xgboost"
 ]
 chem = [
-    "glyles",
+    "glyles~=1.2.3a0",
     "pubchempy",
     "requests",
     "py3Dmol"

